src/auction.js

```diff
@@ -514,68 +528,54 @@ export function getStandardBidderSettings(mediaType) {
-            return bidResponse.pbDg;
-          } else if (granularity === CONSTANTS.GRANULARITY_OPTIONS.LOW) {
-            return bidResponse.pbLg;
-          } else if (granularity === CONSTANTS.GRANULARITY_OPTIONS.MEDIUM) {
-            return bidResponse.pbMg;
-          } else if (granularity === CONSTANTS.GRANULARITY_OPTIONS.HIGH) {
-            return bidResponse.pbHg;
-          } else if (granularity === CONSTANTS.GRANULARITY_OPTIONS.CUSTOM) {
-            return bidResponse.pbCg;
-          }
-        }
-      }, {
-        key: CONSTANTS.TARGETING_KEYS.SIZE,
-        val: function (bidResponse) {
-          return bidResponse.size;
-        }
-      }, {
-        key: CONSTANTS.TARGETING_KEYS.DEAL,
-        val: function (bidResponse) {
-          return bidResponse.dealId;
-        }
-      },
-      {
-        key: CONSTANTS.TARGETING_KEYS.SOURCE,
-        val: function (bidResponse) {
-          return bidResponse.source;
+      createKeyVal(TARGETING_KEYS.BIDDER, 'bidderCode'),
+      createKeyVal(TARGETING_KEYS.AD_ID, 'adId'),
+      createKeyVal(TARGETING_KEYS.PRICE_BUCKET, function(bidResponse) {
+        if (granularity === CONSTANTS.GRANULARITY_OPTIONS.AUTO) {
+          return bidResponse.pbAg;
+        } else if (granularity === CONSTANTS.GRANULARITY_OPTIONS.DENSE) {
+          return bidResponse.pbDg;
+        } else if (granularity === CONSTANTS.GRANULARITY_OPTIONS.LOW) {
+          return bidResponse.pbLg;
+        } else if (granularity === CONSTANTS.GRANULARITY_OPTIONS.MEDIUM) {
+          return bidResponse.pbMg;
+        } else if (granularity === CONSTANTS.GRANULARITY_OPTIONS.HIGH) {
+          return bidResponse.pbHg;
+        } else if (granularity === CONSTANTS.GRANULARITY_OPTIONS.CUSTOM) {
+          return bidResponse.pbCg;
         }
-      },
-      {
-        key: CONSTANTS.TARGETING_KEYS.FORMAT,
-        val: function (bidResponse) {
-          return bidResponse.mediaType;
-        }
-      },
+      }),
+      createKeyVal(TARGETING_KEYS.SIZE, 'size'),
+      createKeyVal(TARGETING_KEYS.DEAL, 'dealId'),
+      createKeyVal(TARGETING_KEYS.SOURCE, 'source'),
+      createKeyVal(TARGETING_KEYS.FORMAT, 'mediaType'),
     ]
 
     if (mediaType === 'video') {
-      [CONSTANTS.TARGETING_KEYS.UUID, CONSTANTS.TARGETING_KEYS.CACHE_ID].forEach(item => {
-        bidderSettings[CONSTANTS.JSON_MAPPING.BD_SETTING_STANDARD][CONSTANTS.JSON_MAPPING.ADSERVER_TARGETING].push({
-          key: item,
-          val: function val(bidResponse) {
-            return bidResponse.videoCacheKey;
-          }
-        })
+      // Adding hb_uuid + hb_cache_id
+      [TARGETING_KEYS.UUID, TARGETING_KEYS.CACHE_ID].forEach(targetingKey => {
+        bidderSettings[CONSTANTS.JSON_MAPPING.BD_SETTING_STANDARD][CONSTANTS.JSON_MAPPING.ADSERVER_TARGETING].push(createKeyVal(targetingKey, 'videoCacheKey'));
       });
+
+      // Adding hb_cache_host + hb_cache_path
```

Comment:
So if mediaType is video,

And cache.url is defined,

Every bidResponse object will now contain `hb_cache_host` and `hb_cache_path` in the `adServerTargeting`.

Sounds good to me.

And if an adapter had also previously specified these two values in its bidResponseObject, then we take it as is instead of overriding with the client side cache.url.

This satisfies the scenario where some bidders use a server side cache and others use client side.

LGTM.